bundles/org.openhab.binding.iaqualink/src/main/java/org/openhab/binding/iaqualink/internal/handler/IAqualinkHandler.java

```diff
@@ -0,0 +1,450 @@
+        String confApiKey = configuration.apiKey;
+
+        if (StringUtils.isBlank(username)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "username must not be empty");
+            return;
+        }
+
+        if (StringUtils.isBlank(password)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "password must not be empty");
+            return;
+        }
+
+        if (StringUtils.isNotBlank(confApiKey)) {
+            this.apiKey = confApiKey;
+        } else {
+            this.apiKey = DEFUALT_API_KEY;
+        }
+
+        this.refresh = configuration.refresh.intValue() > MIN_REFRESH_SECONDS ? configuration.refresh.intValue()
+                : MIN_REFRESH_SECONDS;
+
+        try {
+            AccountInfo accountInfo = client.login(username, password, apiKey);
+            sessionId = accountInfo.getSessionId();
+            if (sessionId == null) {
+                throw new IOException("Response from controller not valid");
+            }
+            logger.debug("SessionID {}", sessionId);
+
+            Device[] devices = client.getDevices(apiKey, accountInfo.getAuthenticationToken(), accountInfo.getId());
+
+            if (devices.length == 0) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No registered devices found");
+                return;
+            }
+
+            if (StringUtils.isNotBlank(confSerialId)) {
+                serialNumber = confSerialId.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
+                if (!Arrays.stream(devices).anyMatch(device -> device.getSerialNumber().equals(serialNumber))) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "No Device for given serialId found");
+                    return;
+                }
+            } else {
+                serialNumber = devices[0].getSerialNumber();
+            }
+
+            initPolling(0);
+
+        } catch (InterruptedException | IOException e) {
+            logger.debug("Could not connect to service");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (NotAuthorizedException e) {
+            logger.debug("Credentials not valid");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Credentials not valid");
+        }
+    }
+
+    /**
+     * Starts/Restarts polling with an initial delay. This allows changes in the poll cycle for when commands are sent
+     * and we need to poll sooner then the next refresh cycle.
+     */
+    private synchronized void initPolling(int initalDelay) {
+        clearPolling();
+        pollFuture = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                pollController();
+            } catch (Exception e) {
+                logger.debug("Exception during poll", e);
```

Comment:
AFAIK we will log it one a global level in the core, so this not needed and you just pass `this::pollController` instead